pkg/ipcache/cidr.go

```diff
@@ -128,6 +132,8 @@ func cidrLabelToPrefix(id *identity.Identity) (prefix netip.Prefix, ok bool) {
 // that were not already upserted. If any 'usedIdentities' are upserted, these
 // are counted separately as they may provide an indication of another logic
 // error elsewhere in the codebase that is causing premature ipcache deletions.
+//
+// DEPRECATED. Prefer UpsertLabels() or OverrideIdentity() instead.
 func (ipc *IPCache) UpsertGeneratedIdentities(newlyAllocatedIdentities map[netip.Prefix]*identity.Identity, usedIdentities []*identity.Identity) {
```

Comment:
I guess the thing to note here is that the incremental policy update path (FQDN) still requires this function until we implement an equivalent mechanism, but that's tracked by #21142.

Reply:
Oh yeah, this is deprecation, not deletion. There's still a whole bunch of users of these existing APIs. The goal here is to (a) discourage using older APIs for new code and (b) encourage discussion around how to remove usage of the older APIs via #21142.